airflow/models/taskinstance.py

```diff
@@ -1135,7 +1136,59 @@ def _run_raw_task(
         if not test_mode:
             session.add(Log(self.state, self))
             session.merge(self)
+
         session.commit()
+        if conf.getboolean('scheduler', 'schedule_after_task_execution', fallback=True):
+            from airflow.models.dagrun import DagRun  # Avoid circular import
+
+            try:
+                # Re-select the row with a lock
+                dag_run = with_row_locks(session.query(DagRun).filter_by(
+                    dag_id=self.dag_id,
+                    execution_date=self.execution_date,
+                )).one()
+
+                # Get a partial dag with just the specific tasks we want to
+                # examine. In order for dep checks to work correctly, we
+                # include ourself (so TriggerRuleDep can check the state of the
+                # task we just executed)
+                tasks = self.task.downstream_list + [self.task]
+                task_ids = [t.task_id for t in tasks]
+
+                partial_dag = self.task.dag.partial_subset(
+                    task_ids,
+                    include_downstream=True,
```

Comment:
Hmmmm, this is going to include _all_ downstream tasks, recursively, not just the tasks we've selected.